inductiva/tasks/task.py

```diff
@@ -175,11 +176,35 @@ def wait(self, polling_period: int = 5) -> models.TaskStatusCode:
     def kill(self) -> None:
         """Kill the task.
 
-        This method issues a request to the API to kill the task. It doesn't
-        block waiting for confirmation if the task was killed.
+        This method issues a request to the API to kill the task.
+        Then waits for 5s (checking every second) to see if the task was killed.
+        If the task is still running (or has a status different from killed)
+        after 5s, it issues a warning.
         """
         self._api.kill_task(path_params=self._get_path_params())
 
+        task_status = self.get_status()
+
+        start_time = time.time()
+
+        while task_status == models.TaskStatusCode.PENDINGKILL:
+
+            time.sleep(1)
+            task_status = self.get_status()
+            if time.time() - start_time > constants.MAX_TIME_WAITING_FOR_TASK:
+                logging.warning(
+                    "Task may not have been killed successfully. Task status: "
+                    "%s", task_status)
+                return
```

Comment:
I am unsure about this. What happens if we want to kill 100 tasks:

```python
tasks_to_kill = [task_1, task_2, ...]
for task in tasks_to_kill:
    task.kill()
```

Will this hang until each task exists the `PENDINGKILL`? The API should be particularly good in handling large workflows, so this seems strange.

If the main issue is repeated `task.kill()` calls to the same task why not just:

```python
if task.status == models.TaskStatusCode.PENDINGKILL:
    logging.warning("Trying to kill task twice")
```

Reply:
It should time out after 5 seconds. That is the maximum time we wait (we can decrease that value).

If the issue is the fact that this can take too long, I can decrease both the sleep timer and the timeout.
Most of the time, we won't even enter the while loop. But sometimes it happens.

Do you think that doing a task.kill sleep and another task.kill would be enough?